iXBRLViewerPlugin/viewer/src/js/inspector.js

```diff
@@ -34,1067 +34,1061 @@ import { DIMENSIONS_KEY, DocumentSummary, MEMBERS_KEY, PRIMARY_ITEMS_KEY, TOTAL_
+                    itemHTML.mouseleave(() => Object.values(r.facts).forEach(f => this._viewer.clearLinkedHighlightFact(f)));
+                    Object.values(r.facts).forEach(f => this._viewer.highlightRelatedFact(f));
+                }
             }
-        });
-        $("<div></div>").addClass("item").addClass("total")
-            .append($("<span></span>").addClass("weight"))
-            .append($("<span></span>").addClass("concept-name").text(fact.getLabelOrName("std")))
-            .appendTo(calcBody);
-
-        a.addCard($("<span></span>").text(label), calcBody, e == elr);
+            $("<div></div>").addClass("item").addClass("total")
+                .append($("<span></span>").addClass("weight"))
+                .append($("<span></span>").addClass("concept-name").text(fact.getLabelOrName("std")))
+                .appendTo(calcBody);
 
-    });
-    return a.contents();
-}
+            a.addCard($("<span></span>").text(label), calcBody, e == elr);
 
-Inspector.prototype._footnotesHTML = function (fact) {
-    var html = $("<div></div>").addClass("fact-list");
-    $.each(fact.footnotes(), (n, fn) => {
-        if (fn instanceof Footnote) {
-            $("<div></div>")
-                .addClass("block-list-item")
-                .text(truncateLabel(fn.textContent(), 120))
-                .mouseenter(() => this._viewer.linkedHighlightFact(fn))
-                .mouseleave(() => this._viewer.clearLinkedHighlightFact(fn))
-                .click(() => this.selectItem(fn.id))
-                .appendTo(html);
         }
-        else if (fn instanceof Fact) {
-            html.append(this.factListRow(fn));
+        return a.contents();
+    }
+
+    _footnotesHTML(fact) {
+        const html = $("<div></div>").addClass("fact-list");
+        for (const fn of fact.footnotes()) {
+            if (fn instanceof Footnote) {
+                $("<div></div>")
+                    .addClass("block-list-item")
+                    .text(truncateLabel(fn.textContent(), 120))
+                    .mouseenter(() => this._viewer.linkedHighlightFact(fn))
+                    .mouseleave(() => this._viewer.clearLinkedHighlightFact(fn))
+                    .click(() => this.selectItem(fn.id))
+                    .appendTo(html);
+            }
+            else if (fn instanceof Fact) {
+                html.append(this.factListRow(fn));
+            }
         }
-    });
-    return html;
-}
+        return html;
+    }
 
-Inspector.prototype.viewerMouseEnter = function (id) {
-    $('.calculations .item').filter(function () {   
-        return $.inArray(id, $.map($(this).data('ivid'), f => f.id )) > -1 
-    }).addClass('linked-highlight');
-    $('#inspector .search .results tr').filter(function () {   
-        return $(this).data('ivid') == id;
-    }).addClass('linked-highlight');
-}
+    viewerMouseEnter(id) {
+        $('.calculations .item')
+            .filter((i, e) => Object.keys($(e).data('ivid') ?? {}).includes(id))
```

Comment:
Can `Object.keys(o).includes(k)` be simplified to `k in o`?

To be sure, this expression doesn't seem equivalent to the other one, which seemed to be calling `f.id` on each entry of `ivid` (so it's an array of objects with `id` attributes).  Is this change fixing or introducing a bug?  I'm getting lost trying to evaluate `resolvedCalculation`, so I'm not sure if `itemHTML.data('ivid', r.facts);` is setting a list of objects or ids.  Perhaps some more doc there would help.

Reply:
> Can `Object.keys(o).includes(k)` be simplified to `k in o`?

Yes, that would be better.

> To be sure, this expression doesn't seem equivalent to the other one, which seemed to be calling `f.id` on each entry of `ivid` (so it's an array of objects with `id` attributes). Is this change fixing or introducing a bug? 

It's not doing the same thing, but should have the same end result.  `r.facts` is a map of fact IDs to fact objects, so the key is the same as the value's `id` property.  I'm really struggling to understand what I was thinking when I wrote the original, as the only reason I can think of for using this data structure rather than an array would have been do an efficient check if a given ID is present.

> I'm getting lost trying to evaluate `resolvedCalculation`, so I'm not sure if `itemHTML.data('ivid', r.facts);` is setting a list of objects or ids. Perhaps some more doc there would help.

Yes - I'll add some further comments.  I don't think we ever actually use the object values of the `ivid` property, so this could just be a list of IDs, and given a better name (e.g. `ivids`) as we use `ivid` to hold a single ID elsewhere.

I've got another long-standing branch for Calc 1.1 which already rewrites a lot of this code, so for this PR I'll just add some comments to better document the current behaviour.

Reply:
I did not realize that `$.map(object, f)` worked on object properties, I thought it was only for arrays.  Now I think I understand how it's working.  But I think `ivid` is always set to `list<string>` elsewhere, so it's confusing that it's set to `map<string,object>` for these calcs.

Reply:
Agreed, it was very confusing.

I've now changed this usage to be a list, and renamed it to `ivids`.